src/main/java/com/tlswe/awsmock/ec2/servlet/MockEc2EndpointServlet.java

```diff
@@ -44,12 +46,25 @@ protected final void doGet(final HttpServletRequest request,
          */
         Map<String, String[]> queryParams = (Map<String, String[]>) request
                 .getParameterMap();
+        Map<String, String> headers = new HashMap<String, String>();
 
-        response.setContentType("text/xml");
-        response.setCharacterEncoding("UTF-8");
+        // To make sure, No response data impacted by others subsequent and concurrent request.
+        synchronized (this) {
```

Comment:
Didn't note that could make aws-mock lose its nature of concurrent request processing a little. However it would be fine for now and we can do another refactor in the handler to change it state-less later and remove this `synchronized`.